Add WebDAV to API2:

jdrossl opened the pull request:
- Add new controller to expose the existing service
- Small refactor for request constants used by other services

https://github.com/craftercms/craftercms/issues/3420

jdrossl committed:
Add WebDAV to API2:
- Add new controller to expose the existing service
- Small refactor for request constants used by other services

sumerjabri commented on src/main/api/2/studio-api-2.yaml:
```diff
@@ -1325,6 +1327,105 @@ paths:
+          required: false
+          schema:
+            type: string
+        - name: type
```
Either switch to MIME-type or spec out the types.

sumerjabri commented on src/main/api/2/studio-api-2.yaml:
```diff
@@ -1325,6 +1327,105 @@ paths:
+                    type: array
+                    items:
+                      $ref: '#/components/schemas/WebDavItem'
+        '401':
```
Add 404

jdrossl replied:
Actually the WebDAV client never fails, it just returns an empty list

jdrossl committed:
Merge branch 'develop' of https://github.com/craftercms/studio into feature/3420

jdrossl committed:
Add new WebDAV service v2:
- Uses shared classes from commons
- Generates remote-assets URLs for files

sumerjabri reviewed (changes requested):
Please add permissions per our convo.

jdrossl committed:
Upgrade Manager:
- Improve config pipelines to avoid warnings for missing files
- Fix the version on some files in the built-in blueprints
- Add new permissions for WebDAV

jdrossl committed:
Update API with WebDAV permissions

jdrossl committed:
Remove unused imports

jdrossl committed:
Fix config for headless blueprints

jdrossl committed:
Merge branch 'develop' of https://github.com/craftercms/studio into feature/3420
